kstatus/status/core.go

```diff
@@ -0,0 +1,486 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package status
```

Comment:
Would you add tests for each of these conditions? -- ideally describe the tests as a list of strings which are the json for the objects, and then the expected results.

Reply:
There are pretty extensive tests for this in status_compute_test.go.